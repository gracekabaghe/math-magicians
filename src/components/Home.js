import React from 'react';

function Home() {
  return (
    <div className="homePage">
      <h1>Welcome</h1>
      <p>
        Howdy! Glad you are here! Feel free to navigate this site and be sure to solve some
        math problems as you look around.
      </p>
      <p>
        There are three pages on this site.
        On the menu click on Quotes to read some insightful things other mathematicians have said.
      </p>
      <p>
        The calculator page contains the main calculator, solve anything that your heart desires.
        Let me know if you have any questions.
      </p>
    </div>
  );
}

export default Home;
