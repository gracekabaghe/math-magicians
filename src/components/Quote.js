import React from 'react';

function Quote() {
  return (
    <div className="quotePage">
      <h1>Quotes</h1>
      <blockquote>
        Pure mathematics is, in its way, the poetry of logical ideas.
        — Albert Einstein, German theoretical physicist
      </blockquote>
      <blockquote>
        Mathematics is the most beautiful and most powerful creation of the human spirit.
        — Stefan Banach, Polish mathematician
      </blockquote>

      <blockquote>
        We will always have STEM with us. Some things will drop out of the public eye and go away,
        but there will always be science, engineering, and technology. And there will always,
        always be mathematics.
        — Katherine Johnson, African-American mathematician
      </blockquote>

    </div>
  );
}

export default Quote;
