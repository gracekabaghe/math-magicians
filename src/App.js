import React, { Component } from 'react';
import './index.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

import Calculator from './components/calculator';
import Quote from './components/Quote';
import Navbar from './components/Navbar';
import Home from './components/Home';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Router>
        <Navbar />
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route path="/calculator">
            <Calculator />
          </Route>
          <Route path="/quote">
            <Quote />
          </Route>
          <Route path="*">
            <Home to="/" />
          </Route>
        </Switch>
      </Router>
    );
  }
}

export default App;
