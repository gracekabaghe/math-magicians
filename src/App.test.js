import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Calculator from './components/calculator';
import Quote from './components/Quote';
import Home from './components/Home';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Math Magicians/i);
  expect(linkElement).toBeInTheDocument();
});
test('change screen when we click navbar on calculator', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('change screen when we click navbar on Quote', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('change screen when we click navbar on Home', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
