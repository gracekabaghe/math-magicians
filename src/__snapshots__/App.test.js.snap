// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change screen when we click navbar on Home 1`] = `
<div
  className="homePage"
>
  <h1>
    Welcome
  </h1>
  <p>
    Howdy! Glad you are here! Feel free to navigate this site and be sure to solve some math problems as you look around.
  </p>
  <p>
    There are three pages on this site. On the menu click on Quotes to read some insightful things other mathematicians have said.
  </p>
  <p>
    The calculator page contains the main calculator, solve anything that your heart desires. Let me know if you have any questions.
  </p>
</div>
`;

exports[`change screen when we click navbar on Quote 1`] = `
<div
  className="quotePage"
>
  <h1>
    Quotes
  </h1>
  <blockquote>
    Pure mathematics is, in its way, the poetry of logical ideas. — Albert Einstein, German theoretical physicist
  </blockquote>
  <blockquote>
    Mathematics is the most beautiful and most powerful creation of the human spirit. — Stefan Banach, Polish mathematician
  </blockquote>
  <blockquote>
    We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics. — Katherine Johnson, African-American mathematician
  </blockquote>
</div>
`;

exports[`change screen when we click navbar on calculator 1`] = `
<div
  className="container"
>
  <div
    className="cal-p"
  >
    <p>
      Solve some Maths
    </p>
  </div>
  <div
    className="calculator-flex"
  >
    <div
      className="output"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button"
        id="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
